File: JavaRepo/lab2/src/model/Statement/IfStmt.java
package model.Statement;

import exceptions.DivideByZeroException;
import exceptions.InvalidOperandException;
import exceptions.NotExistingException;
import model.Expressions.Exp;
import utils.*;
import utils.adt.MyIDictionary;
import utils.adt.MyIHeap;
import utils.adt.MyIStack;

public class IfStmt implements IStmt {
    private Exp exp;
    private IStmt thenS;
    private IStmt elseS;

    public IfStmt(Exp e, IStmt t, IStmt el){
        exp = e;
        thenS = t;
        elseS = el;
    }

    @Override
    public String toString(){
        return "IF("+exp.toString()+")THEN("+thenS.toString()+")ELSE("+elseS.toString()+")";
    }

    @Override
    public PrgState execute(PrgState state) throws DivideByZeroException, InvalidOperandException, NotExistingException {
        MyIDictionary<String,Integer> symbolT = state.getSymTable();
        MyIStack<IStmt> exStack = state.getExeStack();
        MyIHeap<Integer, Integer> heap = state.getHeap();
        int res = exp.eval(symbolT, heap);
        if (res != 0) {
            exStack.push(thenS);
        }
        else {
            exStack.push(elseS);
        }
        //return state;

        return null;
    }
}

File: Year 2/Semester 1/MAP/C# - Toy language/Lab10/Toy Language/Toy Language/model/statement/IfStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language.model.expressions;
using Toy_Language.utils;

namespace Toy_Language.model.statement
{
    public class IfStmt : IStmt
    {
        private Exp exp;
        private IStmt thenS, elseS;

        public IfStmt(
            Exp _exp,
            IStmt _stmt1, 
            IStmt _stmt2
            )
        {
            this.exp = _exp;
            this.thenS = _stmt1;
            this.elseS = _stmt2;
        }

        public PrgState Execute(PrgState state)
        {
            IModDictionary<string, int> symbTable = state.SymbTable;
            IExeStack<IStmt> exeStack = state.ExeStack;

            int res = exp.Eval(symbTable);
            
            if (res != 0)
            {
                exeStack.Push(thenS);
            }
            else
            {
                exeStack.Push(elseS);
            }

            return state;
        }

        public override string ToString()
        {
            return "IF(" + exp.ToString() + ")THEN(" + thenS.ToString() + ")ELSE(" + elseS.ToString() + ")";
        }
    }
}
